File: backend/app/api/v1/crop.py
from fastapi import APIRouter
from app.schemas.crop import CropPredictionRequest
from app.schemas.common import APIResponse
from app.services.crop_service import CropService

router = APIRouter()

@router.post("/predict", response_model=APIResponse)
async def predict(data: CropPredictionRequest):
    result = await CropService.predict(data)
    return APIResponse(data=result)

@router.get("/history", response_model=APIResponse)
async def history():
    return APIResponse(data=[])

File: frontend/src/api/crop.ts
import apiClient from './client';
import type { CropPredictionRequest, CropPredictionResponse } from '../types/crop';

export const predictCrop = async (data: CropPredictionRequest): Promise<CropPredictionResponse> => {
  const response = await apiClient.post('/crop/predict', data);
  return response.data?.data || response.data;
};

export const getCropHistory = async () => {
  const response = await apiClient.get('/crop/history');
  return response.data?.data || response.data;
};
